### PR TITLE
Add YACHT to bioconda

### PR DESCRIPTION
**Pull request by mohsenht**
Describe your pull request here

Add Yacht project to Bioconda. YACHT is a mathematically rigorous hypothesis test for the presence or absence of organisms in a metagenomic sample, based on average nucleotide identity (ANI). Please find the development URL here: https://github.com/KoslickiLab/YACHT

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/yacht/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{% set version = "1.0" %}
+
+package:
+  name: yacht
+  version: {{ version }}
+
+source:
+  url: https://github.com/KoslickiLab/YACHT/releases/download/v1.0/yacht-{{ version }}-py_0.tar.bz2
```
Comment (thanhleviet) [inline recipes/yacht/meta.yaml:8]: https://github.com/KoslickiLab/YACHT/releases/download/`v1.0`/yacht-{{ version }}-py_0.tar.bz
Can `v.1.0` be replaced by {{ version }} too ?

File: recipes/yacht/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  imports:
+    - yacht
+
+  commands:
+    - yacht --help
+
+about:
+  home: https://github.com/KoslickiLab/YACHT
+  summary: 'YACHT is a mathematically rigorous hypothesis test for the presence or absence of organisms in a metagenomic sample, based on average nucleotide identity (ANI).'
+  license: MIT License
```
Comment (thanhleviet) [inline recipes/yacht/meta.yaml:30]: use Short Identifier pls, see https://spdx.org/licenses/MIT.html

**Review by mohsenht — COMMENTED** [review]
Hi thanhleviet,
Thanks for your comments, I resolved them.

File: recipes/yacht/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - yacht
+
+  commands:
+    - yacht --help
+
+about:
+  home: https://github.com/KoslickiLab/YACHT
+  summary: 'YACHT is a mathematically rigorous hypothesis test for the presence or absence of organisms in a metagenomic sample, based on average nucleotide identity (ANI).'
+  license: MIT
+  license_file: LICENSE.txt
```
Comment (thanhleviet) [inline recipes/yacht/meta.yaml:33]: I think you can remove this line. Test failed due to 
```
ValueError: License file given in about/license_file (/opt/recipe/LICENSE.txt) does not exist in source root dir or in recipe root dir (with meta.yaml)
```

Comment (mohsenht) [reply recipes/yacht/meta.yaml]: Deleted

File: recipes/yacht/post-link.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+echo "Installing pyo3-branchwater via pip..."
+"${PREFIX}/bin/pip" install pyo3-branchwater==0.8.1
```
Comment (bgruening) [inline recipes/yacht/post-link.sh:4]: We can not do that ;)

Please create a recipe for that tool as well at https://github.com/conda-forge/staged-recipes/pulls

File: recipes/yacht/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    - pytest
+    - pytest-cov
+    - loguru
+    - maturin >=1,<2
+    - tqdm
+    - pip
+    - biom-format
+    - pytaxonkit
+    - openpyxl
+    - sourmash_plugin_branchwater
```
Comment (mencian) [inline recipes/yacht/meta.yaml:42]:
```suggestion
    - python >3.6
    - sourmash >=4.8.3,<5
    - scipy
    - numpy
    - pandas
    - scikit-learn
    - codecov
    - pytest
    - pytest-cov
    - loguru
    - tqdm
    - biom-format
    - pytaxonkit
    - openpyxl
    - sourmash_plugin_branchwater
```

**Review by mencian — COMMENTED** [review]
`gcc`, `maturin`, and `pip` should not be needed during runtime.

**Review by mencian — APPROVED** [review]
LGTM, thanks @mohsenht!

**Pull request merged**